### PR TITLE
With segment

### PR DESCRIPTION
**Pull request by mulholo**
RE: https://github.com/zeit/next.js/issues/4587, this pull request improves the with-segment example.

Previously, only SSR page loads were tracked. This pull request adds manual page view logging via `Router.events.on('routeChangeComplete')` in `Page.js`.

There is also a minor bug fix on the textarea to remove a console error.

### CODE REVIEW

**Comment by mulholio** [comment]
Wasn't sure if the `typeof window !== 'undefined'` check was necessary or if this was something that was covered by `Router.on`.

File: examples/with-segment-analytics/components/Page.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import Router from 'next/router'
+import Header from './Header'
+
+// Track client-side page views with Segment
+Router.events.on('routeChangeComplete', url => {
+  if (typeof window !== 'undefined') {
```
Comment (timneutkens) [inline examples/with-segment-analytics/components/Page.js]:
```suggestion
```

File: examples/with-segment-analytics/components/Page.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    window.analytics.page(url)
+  }
```
Comment (timneutkens) [inline examples/with-segment-analytics/components/Page.js]:
```suggestion
```

**Comment by timneutkens** [comment]
Router events are not triggered server-side, so it's safe to remove.

**Comment by timneutkens** [comment]
Could you run `yarn lint-fix` so that the code-style is updated?

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Amazing PR!

**Comment by KyruCabading** [comment]
Hello @timneutkens and @mulholio. This is a really helpful PR! I'm using this to setup Segment in our project. I have a question though on implementation of the Router event listener. 
Is there a specific reason why we don't call the listener in _app.js? Won't it have the same effect and only set the listener once on app load? 
```
Router.events.on('routeChangeComplete', url => {
  window.analytics.page(url)
})
```

**Comment by mulholio** [comment]
@KyruCabading Page is just a component which is used to clean up `_app.js` so shouldn't make a difference

**Pull request merged**